VMEncryption/main/Utils/HandlerUtil.py

```diff
@@ -403,72 +408,108 @@ def redo_current_status(self):
+                message = ""
+                
+            message = filter(lambda c: c in string.printable, message)
+            message = message.encode('ascii', 'ignore')
+            
+            self.log("[StatusReport ({0})] op: {1}".format(latest_seq, operation))
+            self.log("[StatusReport ({0})] status: {1}".format(latest_seq, status))
+            self.log("[StatusReport ({0})] code: {1}".format(latest_seq, status_code))
+            self.log("[StatusReport ({0})] msg: {1}".format(latest_seq, message))
+            
+            tstamp = time.strftime(DateTimeFormat, time.gmtime())
+            stat = [{
+                "version" : self._context._version,
+                "timestampUTC" : tstamp,
+                "status" : {
+                    "name" : self._context._name,
+                    "operation" : operation,
+                    "status" : status,
+                    "code" : status_code,
+                    "formattedMessage" : {
+                        "lang" : "en-US",
+                        "message" : message
+                    }
                 }
             }]
-
-            stat[0]["status"]["substatus"] = substat
-
-            if "VMRestartPending" in encryption_status:
-                stat[0]["status"]["formattedMessage"]["message"] = "OS disk successfully encrypted, please reboot the VM"
-
-        stat_rept = json.dumps(stat)
-        # rename all other status files, or the WALA would report the wrong
-        # status file.
-        # because the wala choose the status file with the highest sequence
-        # number to report.
-        if self._context._status_file:
-            with open(self._context._status_file,'w+') as f:
-                f.write(stat_rept)
+            
+            if self.disk_util:
+                encryption_status = self.disk_util.get_encryption_status()
+                
+                self.log("[StatusReport ({0})] substatus: {1}".format(latest_seq, encryption_status))
+                
+                substat = [{
+                    "name" : self._context._name,
+                    "operation" : operation,
+                    "status" : status,
+                    "code" : status_code,
+                    "formattedMessage" : {
+                        "lang" : "en-US",
+                        "message" : encryption_status
+                    }
+                }]
+                
+                stat[0]["status"]["substatus"] = substat
+                
+                if "VMRestartPending" in encryption_status:
+                    stat[0]["status"]["formattedMessage"]["message"] = "OS disk successfully encrypted, please reboot the VM"
+                    
+            stat_rept = json.dumps(stat)
+            # rename all other status files, or the WALA would report the wrong
+            # # status file.
+            # # because the wala choose the status file with the highest sequence
+            # # number to report.
```

Comment:
I could be wrong on this, but I thought the design of WALA was to choose the sequence number of the operation it has just executed, not necessarily the file with the highest number.   WALA drops an _n_.settings file to the config folder at start of the operation process launch, and then looks for an _n_.status file in the status folder when operation process exits.

Reply:
If an operation for n.settings is executing and an n+1.settings is received in between then the guest agent only reports n+1.status. Hence, any status update from n.settings also needs to go to n+1.status.

Reply:
OK, that makes sense to me now.  Could you add this explanation to the code comment?

Reply:
Added comment